docs/specifications/tests/DNSSEC-TP/dnssec13.md

```diff
@@ -1,150 +1,177 @@
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
+
+## Test procedure
 
 1. Create a DNSKEY query with DO flag set for the apex of the
-   *Child Zone*.
+   *Child Zone* ("DNSKEY Query").
 
 2. Create a SOA query with DO flag set for the apex of the
-   *Child Zone*.
+   *Child Zone* ("SOA Query").
 
 3. Create a NS query with DO flag set for the apex of the
-   *Child Zone*.
+   *Child Zone* ("NS Query").
 
 4. Retrieve all name server IP addresses for the
    *Child Zone* using [Method4] and [Method5] ("NS IP").
 
-5. Repeat the following steps for each name server IP address in *NS IP*:
-
-   1. Send the DNSKEY query over UDP.
-      1. If no DNS response is returned, then output
-         *[NO_RESPONSE]* and go to next name server IP address.
-      2. Else, if the DNS response contains no DNSKEY record in the
-         answer section, then output *[NO_RESPONSE_RRSET]* and go to
-         the next name server IP address.
+5. Create the following empty sets:
+
+   1. Name server IP address and associated DNSKEY algorithm
+      ("Algo not signed DNSKEY").
+   2. Name server IP address and associated DNSKEY algorithm
+      ("Algo not signed SOA").
+   3. Name server IP address and associated DNSKEY algorithm
+      ("Algo not signed NS").
+
+6. For each name server IP in the NS IP set do:
+
+   1. Send *DNSKEY Query* over UDP.
+      1. If no DNS response is returned, then go to next name server IP.
+      2. Else, if the RCODE of the DNS response is not "NOERROR" go to next name
+         server IP.
+      3. Else, if the AA flag of the response is unset, then go to the next name
+         server IP.
+      4. Else, if the DNS response contains no DNSKEY record in the answer
+         section, then go to the next name server IP.
+      3. Else, if the DNS response contains no RRSIG for the DNSKEY RRset, then
+         go top the next name server IP.
       3. Else, do:
-         1. Extract all DNSKEY records ("DNSKEY Records").
-         2. Extract the algorithm numbers from each DNSKEY record
-            ("DNSKEY Algorithms").
-         3. Calculate the key ID for each DNSKEY record
-            ("DNSKEY Key ID").
-         4. Extract all RRSIG records from the response.
-         5. If there is no RRSIG for DNSKEY, then output
-            *[RRSET_NOT_SIGNED]* and go to next name server IP
-            address.
-         6. For each algorithm in *DNSKEY Algorithm* do:
+         1. Extract all DNSKEY records.
+         2. Extract the algorithm numbers from each DNSKEY record and add it to
+            a set ("DNSKEY Algorithm").
```

Comment:
It looks like this set is used in the next section for SOA. So we need to create it for each name server.
I would add:
```
6. For each name server IP in the NS IP set do:
    0. Create an empty set ("DNSKEY Algorithm")
```